pr_agent/algo/ai_handler.py

```diff
@@ -70,9 +76,15 @@ async def chat_completion(self, model: str, temperature: float, system: str, use
             TryAgain: If there is an attribute error during OpenAI inference.
         """
         try:
+            deployment_id = self.deployment_id
+            if get_settings().config.verbosity_level >= 2:
+                logging.debug(
+                    f"Generating completion with {model}"
+                    f"{(' from deployment ' + deployment_id) if deployment_id else ''}"
+                )
             response = await acompletion(
                 model=model,
-                deployment_id=self.deployment_id,
+                deployment_id=deployment_id,
                 messages=[
                     {"role": "system", "content": system},
                     {"role": "user", "content": user}
```

Comment:
**Suggestion:** Consider using a more specific exception type in the 'except' clause. Using 'Exception' can potentially catch and ignore important errors that should not be ignored.
```suggestion
        try:
            deployment_id = self.deployment_id
            if get_settings().config.verbosity_level >= 2:
                logging.debug(
                    f"Generating completion with {model}"
                    f"{(' from deployment ' + deployment_id) if deployment_id else ''}"
                )
            response = await acompletion(
                model=model,
                deployment_id=deployment_id,
                messages=[
                    {"role": "system", "content": system},
                    {"role": "user", "content": user}
                ]
            )
        except APIError as e:
            logging.error(f"Failed to generate completion: {e}")
            raise TryAgain("Failed to generate completion") from e
```